src/main/webapp/app/shared/chart/chart-category-filter.ts

```diff
@@ -0,0 +1,126 @@
+import { ChartFilter } from 'app/shared/chart/chart-filter';
+import { Injectable } from '@angular/core';
+import { ExerciseCategory } from 'app/entities/exercise-category.model';
+
+@Injectable({ providedIn: 'root' })
+export class ChartCategoryFilter extends ChartFilter {
+    exerciseCategories: Set<string>;
+    allCategoriesSelected = true;
+    includeExercisesWithNoCategory = true;
+    exercisesWithoutCategoriesPresent: boolean;
+
+    /**
+     * Collects all categories from the provided exercises
+     * @private
+     */
+    determineDisplayableCategories(courseExercises: any[]): Set<any> {
```

Comment:
Please avoid any if possible! Thank you!

Reply:
I agree, there are too many `any` types in this class.